routes/tanksRouter.js

```diff
@@ -17,39 +17,41 @@ tanksRouter.route('/')
   .options(cors.corsWithOptions, (_, res) => { res.sendStatus(200); })
-  .get(cors.cors, async (req, res) => res.json(await getTanksStats(req.params.accountID)))
+  .get(cors.cors, async (req, res) => res.json(await postTanksSnapshots(req.params.accountID)))
   .post(cors.corsWithOptions, authenticate.verifyUser, async (req, res, next) => {
-    const statsToAdd = await postPlayerSnapshots(req.params.accountID);
-    PlayerStats.findOne({ 'data.accountId': req.params.accountID })
-      .then((playerStats) => {
-        if (!playerStats) {
-          PlayerStats.create(statsToAdd)
-            .then((player) => {
-              player.user = req.user._id;
-              player.save();
+    const statsToAdd = await postTanksSnapshots(req.params.accountID);
+    if (!statsToAdd.status) { // if bad data
+      return res.status(404).json(statsToAdd)
+    };
+    statsToAdd.data = statsToAdd.data.filter((tankStats) => tankStats.tank_id);
+    TankStats.findOne({ 'account_id': req.params.accountID })
+      .then((tankStats) => {
+        if (!tankStats) {
+          TankStats.create(statsToAdd)
+            .then((tanks) => {
               res.statusCode = 200;
               res.setHeader('Content-Type', 'application/json');
-              res.json(player);
-            });
+              res.json(tanks);
+            })
+            .catch((err) => console.log(err));
         } else {
-          const snapshotsDB = playerStats.data.snapshots;
-          const snapshotToAdd = statsToAdd.data.snapshots[0];
-          if (snapshotsDB.at(-1).lastBattleTime !== snapshotToAdd.lastBattleTime) {
-            isSameDay(snapshotsDB.at(-1).lastBattleTime, snapshotToAdd.lastBattleTime) ? 
-              snapshotsDB.splice(-1, 1, snapshotToAdd) : 
-              snapshotsDB.push(snapshotToAdd);
-
-            playerStats.save();
-            res.statusCode = 200;
-            res.setHeader('Content-Type', 'application/json');
-            res.json(playerStats);
-          } else {
-            res.statusCode = 200;
-            res.setHeader('Content-Type', 'application/json');
-            res.json(playerStats);
-          }
+          const dataFromDB = tankStats.data;
+          dataFromDB.map((tankFromDB) => {
+            const snapshotToAdd = statsToAdd.data.find((tankFromStats) => tankFromStats.tank_id === tankFromDB.tank_id);
+            if (!snapshotToAdd) return;
+            if (snapshotToAdd.snapshots[0].lastBattleTime !== tankFromDB.snapshots.at(-1).lastBattleTime) {
+              isSameDay(tankFromDB.snapshots.at(-1).lastBattleTime, snapshotToAdd.snapshots[0].lastBattleTime) ? 
+              tankFromDB.snapshots.splice(-1, 1, snapshotToAdd.snapshots[0]) : 
+              tankFromDB.snapshots.push(snapshotToAdd.snapshots[0]);
+            }
+          });
+          tankStats.save();
+          res.statusCode = 200;
+          res.setHeader('Content-Type', 'application/json');
+          res.json(tankStats);
         }
```

Comment:
The code is simplified for better readability and maintainability. The `await` keyword is used when saving `tankStats` to ensure it completes before sending the response
```suggestion
            if (snapshotToAdd.snapshots[0].lastBattleTime !== tankFromDB.snapshots.at(-1).lastBattleTime) {
              tankFromDB.snapshots.push(snapshotToAdd.snapshots[0]);
            }
              if (isSameDay(tankFromDB.snapshots.at(-1).lastBattleTime, snapshotToAdd.snapshots[0].lastBattleTime)) {
          });
                tankFromDB.snapshots.splice(-1, 1, snapshotToAdd.snapshots[0]);
              } else {
                tankFromDB.snapshots.push(snapshotToAdd.snapshots[0]);
             }
           }
      });  
         await tankStats.save();
         res.status(200).json(tankStats);
    } catch  (err) {
      console.log(err);
    }
```